Specs/Core/loadJsonpSpec.js

```diff
@@ -0,0 +1,40 @@
+/*global defineSuite*/
+defineSuite([
+    'Core/loadJsonp',
+    'Core/RequestErrorEvent'
+], function(
+    loadJsonp,
+    RequestErrorEvent) {
+    "use strict";
+    /*global jasmine,describe,xdescribe,it,xit,expect,beforeEach,afterEach,beforeAll,afterAll,spyOn*/
+
+    it('throws with no url', function() {
+        expect(function() {
+            loadJsonp();
+        }).toThrowDeveloperError();
+    });
+
+    it('returns a promise that resolves when the request loads', function() {
+        var testUrl = 'http://example.invalid/testuri';
+        var promise = loadJsonp(testUrl).then(function(data){
+            expect(data).toEqual({
+                good : 'data'
+            });
+        }).otherwise(function(error) {
+
+        });
+
+        expect(promise).toBeDefined();
+    });
+
+    it('returns a promise that rejects when the request errors', function() {
+        var testUrl = 'http://example.invalid/testuri';
+        var promise = loadJsonp(testUrl).then(function(data){
+
+        }).otherwise(function(error) {
+            expect(error).toBeDefined();
+        });
+
+        expect(promise).toBeDefined();
```

Comment:
Not quite, it's because you already called `then` or `otherwise` on the result returned by `loadJsonp`, so if `loadJsonp` returned undefined an exception would have been generated already and execution would have bailed before getting to this line.